e2e/tauri/specs/signin.e2e.js

```diff
@@ -1,11 +1,48 @@
+const fs = require('fs/promises');
+
 describe('Modeling App', () => {
   it('open the sign in page', async () => {
+    // Clean up previous tests
+    await fs.rm('/tmp/kittycad_user_code', { force: true })
+
     const button = await $('#signin')
     expect(button).toHaveText('Sign in')
-    
+
     // Workaround for .click(), see https://github.com/tauri-apps/tauri/issues/6541
     await button.waitForClickable()
     await browser.execute('arguments[0].click();', button)
-    // TODO: handle auth
+    await new Promise(resolve => setTimeout(resolve, 1000))
+
+    // Get from main.rs
+    const userCode = await (await fs.readFile('/tmp/kittycad_user_code')).toString()
+    console.log(`Found user code ${userCode}`)
+
+    // Device flow: verify
+    const token = process.env.KITTYCAD_API_TOKEN
+    const headers = {
+      Authorization: `Bearer ${token}`,
+      Accept: 'application/json',
+      "Content-Type": "application/json",
+    }
+    const verifyUrl = `https://api.kittycad.io/oauth2/device/verify?user_code=${userCode}`
+    console.log(`GET ${verifyUrl}`)
+    const vr = await fetch(verifyUrl, { headers })
+    console.log(vr.status)
+
+    // Device flow: confirm
+    const confirmUrl = 'https://api.kittycad.io/oauth2/device/confirm'
+    const data = JSON.stringify({ user_code: userCode })
+    console.log(`POST ${confirmUrl} ${data}`)
+    const cr = await fetch(confirmUrl, {
+      headers,
+      method: 'POST',
+      body: data,
+    })
+    console.log(cr.status)
+
+    // Now should be logged in
+    const newFileButton = await $('#new_file')
+    await newFileButton.waitForClickable({ timeout: 30000 })
```

Comment:
I'm a little stuck here, as in the CI the confirm call gets the right 204 success code, but it doesn't seem to move to the home page with 'Get user data...' as logs, like I see on my Ubuntu machine

Reply:
The good news is, I can reproduce on a Github Codespace. Might have something to do with the headless env